### PR TITLE
GCU test for BGPListener(BGPMonitor)

### PR DESCRIPTION
**Pull request by wen587**
### Description of PR

Summary: Testcase of BGPListener(BGPMonitor) for generic updater apply-patch
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
End to End test support for Generic Updater apply-patch
This PR is to verify the usage of 'config apply-patch' works on BGPL

#### How did you do it?
Make some config apply change. And check if the BGPLconfig changed as expected.

#### How did you verify/test it?
Run test of sonic-mgmt/tests/generic_config_updater/test_bgpl.py on KVM

### CODE REVIEW

File: tests/generic_config_updater/test_bgpl.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    cmds = 'sonic-db-cli CONFIG_DB keys "BGP_MONITORS|*" | xargs -r sonic-db-cli CONFIG_DB del'
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'],
+        "bgpmon cleanup config failed"
+    )
+
+def check_bgpmon_with_addr(duthost, addr):
+    """ Check BGP MONITOR config change is taken into effect
+    """
+    cmds = "show ip bgp summary | grep -w {}".format(addr)
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_bgpl.py:61]: >show ip bgp summary [](http://example.com/codeflow?start=12&length=19)

Maybe easier to use `show ip bgp nei {}`

File: tests/generic_config_updater/test_bgpl.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'],
+        "BGPMonitor with addr {} is not being setup.".format(addr)
+    )
+
+def bgpmon_tc1_add_init(duthost, bgpmon_setup_info):
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_bgpl.py:67]: >bgpmon_tc1_add_init [](http://example.com/codeflow?start=4&length=19)

Do we need extra testcases to admin up/down an existing peer?

File: tests/generic_config_updater/test_bgpl.py (hunk 1 above)
Comment (wen587) [reply tests/generic_config_updater/test_bgpl.py:61]: `show ip bgp nei` doesn't work on BGPMonitor.
```
admin@vlab-01:~/bgpl$ show ip bgp sum

IPv4 Unicast Summary:
BGP router identifier 10.1.0.32, local AS number 65100 vrf-id 0
BGP table version 4
RIB entries 3, using 576 bytes of memory
Peers 5, using 109080 KiB of memory
Peer groups 5, using 320 bytes of memory


Neighbhor      V     AS    MsgRcvd    MsgSent    TblVer    InQ    OutQ  Up/Down    State/PfxRcd    NeighborName
-----------  ---  -----  ---------  ---------  --------  -----  ------  ---------  --------------  --------------
10.0.0.57      4  64600          0          0         0      0       0  never      Active          ARISTA01T1
10.0.0.59      4  64600          0          0         0      0       0  never      Active          ARISTA02T1
10.0.0.61      4  64600          0          0         0      0       0  never      Active          ARISTA03T1
10.0.0.63      4  64600          0          0         0      0       0  never      Active          ARISTA04T1
11.0.0.1       4  65100          0          0         0      0       0  never      Active          BGPMonitor

Total number of neighbors 5
admin@vlab-01:~/bgpl$ show ip bgp nei 11.0.0.1
Usage: show ip bgp nei [OPTIONS] [IPADDRESS] [[routes|advertised-
                       routes|received-routes]]
Try "show ip bgp nei -h" for help.

Error:  Bgp neighbor 11.0.0.1 not configured

```

File: tests/generic_config_updater/test_bgpl.py (hunk 2 above)
Comment (wen587) [reply tests/generic_config_updater/test_bgpl.py:67]: Add admin down test for existing bgpmon.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**